src/ContactCard.js

```diff
@@ -1,24 +1,56 @@
 import React, { Component } from 'react';
-import App from './App';
-import ContactList from './ContactList';
+import './Contactcard.css';
+import logo from './logo.svg';
 
 class ContactCard extends Component {
-    constructor(props){
-        super(props);
-        this.state= {card : props.contactData }
-        this.deleteContact= this.deleteContact.bind(this);;
-    }
-    deleteContact(e){
-        this.state.card.pop(this.props.id);
+    
+constructor(props) {
+    super(props);
+    this.state={ cardItem: props.contactData, editMode: false }
+    this.newCard=this.newCard.bind(this);
+    this.formDisplay=this.formDisplay.bind(this);
+}
+
+newCard(e, field)  {
+    var newValue = e.target.value; 
+    this.setState({cardItem : newValue});
+    this.props.editCard(this.state.cardItem.id, newValue, field);
+}
+
+componentWillReceiveProps(nextProps) {
+    this.setState({ cardItem: nextProps.contactData })
+}
+
+formDisplay(e, id){
+    this.setState({editMode : !this.state.editMode})
+}
+
+render() {
+    var cls = 'foredit' + (this.state.editMode ? 'visible' : '')
```

Comment:
Why two different kinds of logics to showHide forms in ContactCard.js and ContactList.js. Try to make both of them uniform inside the same project.